internal/proxyservice/proxy_service.go

```diff
@@ -156,6 +169,38 @@ func (s *proxyServices) createHTTPService(
 	return newSvc, nil
 }
 
+func (s *proxyServices) createSSHService(
```

Comment:
I think we might be stuffing way too much into this file in an almost copy/paste manner making me think this can get collapsed in some way. Can this be split out or genericized to avoid needing extra code here for every type of connector?

Reply:
We definitely should do that but I'd say let's not do it as part of this particular PR. Unless if there's a good abstraction in mind right now

Reply:
Fair but I would expect to see at least a bit of work on this for your next PR if you're touching the same code :)